pkg/traceql/engine_metrics.go

```diff
@@ -1112,6 +1126,14 @@ func (m *MetricsFrontendEvaluator) Results() SeriesSet {
 	m.mtx.Lock()
 	defer m.mtx.Unlock()
 
+	if m.metricsSecondStage != nil {
```

Comment:
I find this logic order a little confusing since it checks second stage first?  

Maybe something like:

```
result := m.metricsPipeline.result()

if m.metricsSecondStage != nil {
result = m.metricsSecondStage.process(result)
}

return result
```

(also in MetricsEvaluator.Result())

Reply:
done